system-variables: add tidb_paging_size_bytes (internal use) (#23565)
pingcap/docs: system-variables.md

- Range: `[0, 2147483647]`
- This variable controls the iteration of the optimizer's estimation logic. After changing the value of this variable, the estimation logic of the optimizer will change greatly. Currently, `0` is the only valid value. It is not recommended to set it to other values.

> ### `tidb_paging_size_bytes` <span class="version-mark">New in v9.0.0</span>

Author: ```suggestion
### `tidb_paging_size_bytes` <span class="version-mark">New in TiDB-X-CLOUD.202603</span>
```

> - Scope: SESSION | GLOBAL

Author: ```suggestion
- Scope: SESSION | GLOBAL for TiDB Self-Managed; SESSION for TiDB Cloud Essential and Premium
```

- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
- Type: Integer
- Default value: `0`
- Range: `[0, 9223372036854775807]`
- Unit: bytes

> - Controls the upper limit on the number of bytes in a single paged response in the coprocessor protocol, providing additional pagination by byte count beyond the row-count-based pagination mechanism provided by [`tidb_max_paging_size`](#tidb_max_paging_size-new-in-v630). The default value is `0`, which means pagination by byte count is disabled. This feature takes effect only when [resource control](/tidb-resource-control-ru-groups.md) is enabled and the resource group to which the current statement belongs has a fixed RU quota. It is used by PD's resource control module to estimate and pre-deduct RU based on the amount of scanned data before executing the statement. If you need to enable it, consider setting it to `4194304` (that is, 4 MiB).

Author: ```suggestion
- Controls the maximum size, in bytes, of a single paged response in the coprocessor protocol. Row-based pagination is controlled by [`tidb_max_paging_size`](#tidb_max_paging_size-new-in-v630), while this variable provides an additional pagination mechanism based on response size. The default value is `0`, which disables byte-based pagination. This feature takes effect only when [resource control](/tidb-resource-control-ru-groups.md) is enabled, and the resource group of the current statement has a fixed RU quota. It allows the PD resource control module to estimate RU consumption based on the amount of data to be scanned and deduct the estimated RUs in advance before the statement is executed. To enable this feature, consider setting the variable to `4194304` (4 MiB).
```

- Range: `[0, 2147483647]`
- This variable controls the iteration of the optimizer's estimation logic. After changing the value of this variable, the estimation logic of the optimizer will change greatly. Currently, `0` is the only valid value. It is not recommended to set it to other values.

### `tidb_paging_size_bytes` <span class="version-mark">New in v9.0.0 and TiDB-X-CLOUD.202603</span>

>**Note:**
>
> This variable is not supported on TiDB Cloud Starter.

> - Scope: SESSION | GLOBAL for TiDB Self-Managed; SESSION for TiDB Cloud Essential and Premium

Author: ```suggestion
- Scope
    - TiDB Self-Managed: SESSION | GLOBAL
    - TiDB Cloud Essential and Premium: SESSION
```
